src/ConnectedMode/ServerSentEvents/SSESessionManager.cs

```diff
@@ -36,10 +36,12 @@ internal interface ISSESessionManager : IDisposable
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal sealed class SSESessionManager : ISSESessionManager
     {
+        private readonly object syncRoot = new object();
```

Comment:
It's a singleton but still better to be explicit and mark it static.

Reply:
The fields can't be static because they're MEF imported.
I think that actually avoiding deadlocks is a problem -- I'd rather get the deadlock and see it so I could fix it.

Reply:
I hard disagree. What if the deadlock happens on a user's machine? How are you going to repro and fix it?

Reply:
It's a minor issue so agree to disagree and keep it however.